### PR TITLE
Cloud azure

### PR DESCRIPTION
**Pull request by s-schweer**
### What does this PR do?

disable annoying failure because of reference to object from optional import as default in function definition
### Tests written?

No

### CODE REVIEW

**Comment by cachedout** [comment]
@techhat What say you?

**Comment by cachedout** [comment]
@s-schweer It looks like the linter is fairly worried about this: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/2047/violations/file/salt/cloud/clouds/azurearm.py/

File: salt/cloud/clouds/azurearm.py [hunk 1]
```diff
@@ -146,11 +146,15 @@ def get_dependencies():
     )
 
 
-def get_conn(Client=ComputeManagementClient,  # pylint: disable=invalid-name
-             ClientConfig=ComputeManagementClientConfiguration):  # pylint: disable=invalid-name
+def get_conn(Client, ClientConfig):
```
Comment (techhat) [inline salt/cloud/clouds/azurearm.py]: Should be `Client=None, ClientConfig=None`.

File: salt/cloud/clouds/azurearm.py [hunk 2]
```diff
@@ -146,11 +146,15 @@ def get_dependencies():
...
     '''
     Return a conn object for the passed VM data
     '''
+    if not Client:
```
Comment (techhat) [inline salt/cloud/clouds/azurearm.py]: `if Client is None:`

File: salt/cloud/clouds/azurearm.py [hunk 3]
```diff
@@ -146,11 +146,15 @@ def get_dependencies():
...
+        Client = ComputeManagementClient
+    if not ClientConfig:
```
Comment (techhat) [inline salt/cloud/clouds/azurearm.py]: `if ClientConfig is None:`

**Comment by techhat** [comment]
I haven't seen the `NameError`, but so long as the changes I recommended are made (which will also clean up the lint errors), I don't see a problem with this.

**Comment by cachedout** [comment]
Many thanks, @techhat 

@s-schweer when you get those changes in then I will merge this. Thanks!

**Comment by s-schweer** [comment]
@techhat i fixed the lint errors, maybe you didn't see the NameError because of # pylint: disable=invalid-name

**Comment by techhat** [comment]
That would have blocked me from seeing it in pylint to be sure, but I didn't see them elsewhere. Regardless, I appreciate the fix. Thanks!

**Comment by s-schweer** [comment]
of course they only appear when the whole python-azure libs are not installed, referencing ComputeManagementClient as default value in the function definition raises the NameError during import, my approach would raise it while running the function...

**Comment by techhat** [comment]
That makes sense.

**Pull request merged**